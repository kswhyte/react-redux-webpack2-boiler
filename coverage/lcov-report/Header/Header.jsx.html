<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Header/Header.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Header</a> Header.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
&nbsp;
import './header.css';
&nbsp;
const Header = props =&gt; {
  Header.propTypes = {
    headerSize: PropTypes.string
  };
&nbsp;
  //handle no margin getting overwritten?
  return (
    &lt;header className={props.headerSize === 'small' ? <span class="branch-0 cbranch-no" title="branch not covered" >'small' </span>: null}&gt;
      &lt;div className="row no-margin"&gt;
        &lt;div className="logo-window col-xs-6"&gt;
          &lt;div className="logo-animation"&gt;
            &lt;div className="logo-full" /&gt;
            &lt;div className="logo-small" /&gt;
          &lt;/div&gt;
        &lt;/div&gt;
&nbsp;
        &lt;div className="col-xs-6"&gt;
          &lt;div className="pull-right"&gt;
            &lt;Link to="/"&gt;
              &lt;i className="fa fa-tachometer fa-2" aria-hidden="true" /&gt;
              &lt;div className="text-small-o"&gt;Dashboard&lt;/div&gt;
            &lt;/Link&gt;
            &lt;Link to="/"&gt;
              &lt;i className="fa fa-cog fa-2" aria-hidden="true" /&gt;
              &lt;div className="text-small-o"&gt;Settings&lt;/div&gt;
            &lt;/Link&gt;
          &lt;/div&gt;
        &lt;/div&gt;
      &lt;/div&gt;
&nbsp;
      &lt;div&gt;
        &lt;div&gt;MR Metropolitan Region&lt;/div&gt;
      &lt;/div&gt;
    &lt;/header&gt;
  );
};
&nbsp;
export default Header;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jul 26 2017 09:19:07 GMT-0600 (MDT)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
