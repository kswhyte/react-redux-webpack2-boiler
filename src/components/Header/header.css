@import "imports/_helpers.css";

header {
  &.small + .body-wrapper {
    margin-top: 145px;
  }

  & + .body-wrapper {
    transition: margin-top 0.5s;
    transition-timing-function: ease-in-out;
  }

  &.small .logo-animation {
    top: -73px;
  }

  &.small .logo-window {
    height: 36px;
  }

  &.small > div:first-of-type {
    height: 36px;
  }

  &.small > div:nth-child(1) a div {
    opacity: 0;
    bottom: -50px;
    height: 18px;
  }
  &.small > div:first-of-type > div:last-of-type a:link i.fa {
    top: 10px;
  }

  & > div {
    padding: 0 93px;
    display: flex;

    & div {
      align-self: center;
    }
  }

  & > div:first-of-type {
    background-color: var(--slate);
    height: 81px;
    transition: height 0.5s;
    transition-timing-function: ease-in-out;

    & > div:last-of-type {
      color: var(--white);

      & a:any-link {
        text-declaration: none;
        color: var(--white);
        display: inline-block;
        text-align: center;
        padding-right: 30px;

        & i {
          font-size: 18px;
          position: relative;
          top: 0px;
          transition: top 0.5s;
        }
      }
    }
  }

  & > div:last-of-type {
    background-color: var(--white);
    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
    height: 70px;
  }

  & > div:nth-child(1) a div {
    opacity: 1;
    position: relative;
    bottom: 0px;
    transition: opacity 1.5s, bottom 1.0s, height 1.0s;
  }

  & .logo-window {
    position: relative;
    height: 81px;
    overflow: hidden;
    transition: height 0.5s;
    transition-timing-function: ease-in-out;
  }

  & .logo-animation {
    position: absolute;
    top: 15px;
    left: 0px;
    transition: top 0.5s;
    transition-timing-function: ease-in-out;
  }

  & .logo-full {
    background-image: image-set(
      url('~/images/hg_smartAccess_logo.png') 1x,
      url('~/images/hg_smartAccess_logo@2x.png') 2x
    );
    width: 250px;
    height: 50px;
    background-repeat: no-repeat;
  }

  & .logo-small {
    background-image: image-set(
      url('~/images/hg_smartAccess_small.png') 1x,
      url('~/images/hg_smartAccess_small@2x.png') 2x
    );
    width: 212px;
    height: 30px;
    margin-top: 30px;
    background-repeat: no-repeat;
  }

  & .manager-logo {
    background-image: image-set(url('~/images/MR_logo.png') 1x, url('~/images/MR_logo@2x.png') 2x);
    width: 212px;
    height: 30px;
    background-repeat: no-repeat;
  }

  & > div:last-of-type .pull-right {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
  }

  & > div:last-of-type .sign-in-status,
  & > div:last-of-type .subheader-pipe,
  & > div:last-of-type a {
    display: inline-block;
  }

  & > div:last-of-type .sign-in-status {
    text-align: right;
    color: var(--light-grey);
  }

  & > div:last-of-type .sign-in-status span {
    font-weight: 600;
  }

  & > div:last-of-type div {
    margin-right: 10px;
  }

  & > div:last-of-type a:any-link {
    font-family: Open Sans;
    text-decoration: none;
    color: var(--link-color);
  }
}
