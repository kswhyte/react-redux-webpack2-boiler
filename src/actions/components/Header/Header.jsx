import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './header.css';

const Header = props => {
  Header.propTypes = {
    headerSize: PropTypes.string
  };

  //handle no margin getting overwritten?
  return (
    <header className={props.headerSize === 'small' ? 'small' : null}>
      <div className="row no-margin">
        <div className="logo-window col-xs-6">
          <div className="logo-animation">
            <div className="logo-full" />
            <div className="logo-small" />
          </div>
        </div>

        <div className="col-xs-6">
          <div className="pull-right">
            <Link to="/">
              <i className="fa fa-tachometer fa-2" aria-hidden="true" />
              <div className="text-small-o">Dashboard</div>
            </Link>
            <Link to="/">
              <i className="fa fa-cog fa-2" aria-hidden="true" />
              <div className="text-small-o">Settings</div>
            </Link>
          </div>
        </div>
      </div>

      <div className="row no-margin">
        <div className="col-xs-12 col-sm-3">
          <div className="manager-logo" />
        </div>

        <div className="col-xs-12 col-sm-9">
          <div className="pull-right">
            <div className="sign-in-status">
              Currently signed in as<br />
              <span>AgentUsername</span>
            </div>
            <div className="subheader-pipe">|</div>
            {/* {make this a tag, below, into a Link and connect to logout path} */}
            <a href="#">Log out</a>
          </div>
        </div>
      </div>
    </header>
  );
};

export default Header;
