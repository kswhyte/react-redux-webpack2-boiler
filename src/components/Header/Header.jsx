import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './header.css';

const Header = props => {
  Header.propTypes = {
    headerSize: PropTypes.string
  };

  //handle no margin getting overwritten?
  return (
    <header className={props.headerSize === 'small' ? 'small' : null}>
      <div className="row no-margin">
        <div className="logo-window col-xs-6">
          <div className="logo-animation">
            <div className="logo-full" />
            <div className="logo-small" />
          </div>
        </div>

        <div className="col-xs-6">
          <div className="pull-right">
            <Link to="/">
              <i className="fa fa-tachometer fa-2" aria-hidden="true" />
              <div className="text-small-o">Dashboard</div>
            </Link>
            <Link to="/">
              <i className="fa fa-cog fa-2" aria-hidden="true" />
              <div className="text-small-o">Settings</div>
            </Link>
          </div>
        </div>
      </div>

      <div>
        <div>MR Metropolitan Region</div>
      </div>
    </header>
  );
};

export default Header;
