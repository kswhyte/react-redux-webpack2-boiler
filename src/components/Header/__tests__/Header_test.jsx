import React from 'react';
// import { storiesOf, action, linkTo } from '@kadira/storybook';
import Header from '../Header';

// import { specs, describe, it } from 'storybook-addon-specifications';

import {storiesOf, describe, it, specs
} from "../../../../.storybook/facade";

import {shallow} from 'enzyme';
import expect from 'expect';


const stories = storiesOf('Header', module);

stories.add('Story', () => {
    const headerStory = (
      <Header/>);

    specs(() => describe('Show a successful alert', () => {
        it('Should render the Header component without crashing', () => {
            let output = shallow(headerStory);
            expect(output.find('header').length).toEqual(1);
        });
    }));
    return headerStory;
});
