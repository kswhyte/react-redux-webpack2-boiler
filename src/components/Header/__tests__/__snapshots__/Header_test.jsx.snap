// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story 1`] = `
<header>
  <div
    className={null}
  >
    <div>
      hg smartaccess
    </div>
  </div>
  <div>
    <div>
      MR Metropolitan Region
    </div>
  </div>
</header>
`;
